addons/html_editor/static/src/main/chatgpt/chatgpt_plugin.js

```diff
@@ -123,21 +172,73 @@ export class ChatGPTPlugin extends Plugin {
-        if (selection.isCollapsed) {
-            this.dependencies.dialog.addDialog(ChatGPTPromptDialog, { ...dialogParams, sanitize });
-        } else {
+        if (params.language) {
             const originalText = selection.textContent() || "";
             this.dependencies.dialog.addDialog(
-                params.language ? ChatGPTTranslateDialog : ChatGPTAlternativesDialog,
+                ChatGPTTranslateDialog,
                 { ...dialogParams, originalText, sanitize }
             );
-        }
+        } else {
+            let recordModel, recordName, recordId, callerId, placeholderPrompt, callerComp, textSelection, frontEndRecordInfo;
+            const { resModel, resId, data, fields, id } = this.config.getRecordInfo();
+            const recordInfoJSON = this.recordDataToJSON(data, fields);
+            if (selection.isCollapsed) {
+                if (resModel === "mail.compose.message") {
+                    callerComp = "html_field_composer";
+                    recordName = data.record_name;
+                    recordModel = data.model;
+                    recordId = Number(data.res_ids.slice(1,-1));  // resIds should look like so `[id]`, the slice and cast allows to extact the id
+                    placeholderPrompt = _t('Write a follow up answer');
+                    callerId = id;
+                } else {
+                    callerComp = "html_field_record";
+                    recordName = recordInfoJSON.name;
+                    recordModel = resModel;
+                    placeholderPrompt = "";
+                    frontEndRecordInfo = JSON.stringify(recordInfoJSON);
+                    callerId = resId || id;
+                    this.services['mail.store'].aiInsertButtonTarget = resId || id;
+                }
+            } else {
+                callerComp = "html_field_text_select";
+                recordName = data.record_name || recordInfoJSON.name;
+                placeholderPrompt = _t("Rewrite");
+                textSelection = selection.textContent();
+                callerId = resId || id;
+                this.services['mail.store'].aiInsertButtonTarget = resId || id;
+            }
+            const ai_channel_id = await this.services.orm.call(
+                'discuss.channel',
+                'create_ai_composer_channel',
+                [ 
+                    callerComp,
+                    recordName,
+                    recordModel,
+                    recordId,
+                    frontEndRecordInfo,
+                    textSelection,
+                ], 
+            );
+            const thread = await this.services['mail.store'].Thread.getOrFetch({
+                model: "discuss.channel",
+                id: Number(ai_channel_id), 
+            });
+            thread.composer.text = placeholderPrompt;
+            thread.aiChatSource = callerId;
+            thread.aiSpecialActions = {
+                'insert': dialogParams.insert,
+            };
```

Comment:
You write a few specific properties on that mail.store thread object, and in other places too. I don't know exactly how that kind of object is supposed to be used so maybe you're using it correctly, but it feels weird to me to write such specific properties on an object that otherwise seems to be rather generic.

Reply:
These are contextual data that are relevant on transversal components.
- Storing in thread is to allow interactions between thread and the html composer
- Storing on `store.aiInsertButtonTarget` is a way to globally tell UI ai composer is being used, which is again relevant for other transversal components to have custom rules based on that, notably only show ai_agent chat windows.